### ISSUE TITLE
[C++] syntax highlighting fails with this line

### ISSUE THREAD

**Issue by polyvertex** [sublimehq/Packages#774]
### Summary

The default C++ syntax package does not seem to handle the following line:
```c++
var = *reinterpret_cast<std::vector<std::shared_ptr<AnyClass>>*>(v);
```

As a result, the remaining of the file (i.e. starting with this line) is:
*  **incorrectly** highlighted (some parts seem to be skipped)
* not accessible via the `Goto Symbol` feature (<kbd>Ctrl+R</kbd>). Unlike symbols that are **before** this line

### Steps to reproduce

1. Open a file
2. Ensure the default `C++` syntax is active
3. Add the aforementioned C++ line wherever it would suit in a normal code
4. Now observe the remaining part of the file

### Environment

* SublimeText 3126 64bit
* Default `C++` package
* Colo scheme: `Monokai` as least (default one)

**Comment by wbond** [comment]
Here is an example that makes the error more clear:

```c++
int main(void) {
	var = *reinterpret_cast<std::vector<std::shared_ptr<AnyClass>>*>();
}
```

**Comment by polyvertex** [comment]
@wbond if I may, your example is syntactically incorrect. A valid expression is missing between the parentheses of the cast.

**Comment by wbond** [comment]
@polyvertex Updated

**Comment by wbond** [comment]
This was fixed by 5b325fa
